dipper/sources/OrthoXML.py

```diff
@@ -0,0 +1,229 @@
+        :return: None
+
+        """
+        self.get_files(is_dl_forced)
+
+    def parse(self, limit=None):
+        """
+        :return: None
+        """
+
+        if self.testOnly:
+            self.testMode = True
+
+        if self.tax_ids is None:
+            logger.info(
+                "No taxon filter set; Dumping all orthologous associations.")
+        else:
+            logger.info(
+                "Only the following taxa will be dumped: %s",
+                str(self.tax_ids))
+
+        self._get_relations(limit)
+
+        return
+
+    def _get_relations(self, limit):
+        """
+        This will process each of the specified orthoxml files, and
+        extracting the induced orthology and paralogy associations
+        based on the specified xml group nodes.
+
+        The specs for orthoxml can be found here: http://orthoxml.org
+
+        We currently extract tripples for orthologous relations,
+        paralogous relations and in_taxon relations to NCBITaxonId
+        attributes, e.g.
+
+        Triples:
+        <protein1_id> RO:othologous <protein2_id>
+        <assoc_id> :hasSubject <protein1_id>
+        <assoc_id> :hasObject <protein2_id>
+        <assoc_id> :hasPredicate <RO:orthologous>
+        <assoc_id> dc:evidence ECO:phylogenetic_evidence
+
+        :param limit: limit the number of induced pairwise relations
+        :return: None
+
+        """
+        logger.info("getting orthologs and paralog relations")
+
+        g = self.testgraph if self.testMode else self.graph
+        model = Model(g)
+
+        for k in self.files.keys():
+            f = os.path.join(self.rawdir, self.files[k]['file'])
+            matchcounter = 0
+            logger.info("Parsing %s", f)
+
+            xml = lxml.etree.parse(f)
+            parser = OrthoXMLParser(xml)
+
+            for protein_nr_a, protein_nr_b, rel_type in parser.extract_pairwise_relations():
+                protein_a = parser.gene_mapping[protein_nr_a]
+                protein_b = parser.gene_mapping[protein_nr_b]
+
+                protein_id_a = protein_a.get('protId')
+                protein_id_b = protein_b.get('protId')
+
+                if self.testMode and not \
+                        (protein_id_a in self.test_ids or protein_id_b in self.test_ids):
+                    continue
+
+                matchcounter += 1
+                taxon_a = self.extract_taxon_info(protein_a)
+                taxon_b = self.extract_taxon_info(protein_b)
+
+                rel = self._map_orthology_code_to_RO[rel_type]
+                evidence_id = 'ECO:0000080'  # phylogenetic evidence
+
+                # add genes to graph;
+                # assume labels will be taken care of elsewhere
+                protein_id_a = self.clean_protein_id(protein_id_a)
+                protein_id_b = self.clean_protein_id(protein_id_b)
+                model.addClassToGraph(protein_id_a, None)
+                model.addClassToGraph(protein_id_b, None)
```

Comment:
Can we type these as SO:0000104? 
i.e. model.addClassToGraph(protein_id_a, None, "SO:0000104")
or 
model.addClassToGraph(protein_id_a, None, Genotype.genoparts['polypeptide'])

@mbrush does it matter if these are individuals or classes?

Reply:
yes, seems like SO:0000104 is exactly what it is. will update accordingly. Thanks for pointing out